recipients_app/lib/data/repositories/organization_repository.dart

```diff
@@ -1,24 +1,27 @@
+import "package:app/data/datasource/demo/organization_demo_data_source.dart";
+import "package:app/data/datasource/organization_data_source.dart";
+import "package:app/data/datasource/remote/organization_remote_data_source.dart";
 import "package:app/data/models/organization.dart";
+import "package:app/demo_manager.dart";
 import "package:cloud_firestore/cloud_firestore.dart";
 
 class OrganizationRepository {
+  late OrganizationDataSource remoteDataSource = OrganizationRemoteDataSource(firestore: firestore);
```

Comment:
Try to avoid late when possible. Instead its better to pass the remoteDataSource and localDataSource via dependency injection:

```
class OrganizationRepository {
    final OrganizationDataSource remoteDataSource;
    final OrganizationDataSource demoDataSource;
    ...

    const OrganizationRepository({
    required this.remoteDataSource,
    required this.demoDataSource,
    ...
  });

}
```

Reply:
@novas1r1 
But would not save the current implementation memory e.g. in the case if a user only uses the demo mode? Then the instance for "remoteDataSource" is never created and does not use memory.
So, could you explain a bit more why it is better to avoid late? Would be interesting :)